### PR TITLE
C ext for correctness - WIP

### PR DESCRIPTION
**Pull request by byroot**

### CODE REVIEW

File: lib/bootscale/features.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    def dump(features)
+      features.join("\0")
+    end
+
+    def load(payload)
+      payload.split("\0")
+    end
+  end
+end
```
Comment (grosser) [inline lib/bootscale/features.rb:30]: no newline ;)

**Comment by grosser** [comment]
funky shit ... a test that fails without these correctness fixes would be great :)

**Comment by byroot** [comment]
Actually we figured a couple hours ago with @csfrancis, that this approach can't work.

Most of the internal functions we'd need to call like `rb_feature_p` are not exposed to C extensions :/

So right now I can't think of any way to reach perfect compatibility with the original Kernel#require. We'd need to work on MRI itself, with no guaranties it would be accepted.

:cry:

**Comment by grosser** [comment]
should be a PR to mri that just fills the cache when a new loap-path is
added ... then we don't need bootscale anymore ...

**Pull request closed without merging**